imblearn/base.py

```diff
@@ -254,105 +158,17 @@ def __setstate__(self, dict):
         self.logger = logger
 
 
-class BaseBinarySampler(six.with_metaclass(ABCMeta, SamplerMixin)):
-    """Base class for all binary class sampler.
-
-    Warning: This class should not be used directly. Use derived classes
-    instead.
-
-    """
-
-    def __init__(self, ratio='auto', random_state=None):
-        """Initialize this object and its instance variables.
-
-        Parameters
-        ----------
-        ratio : str or float, optional (default='auto')
-            If 'auto', the ratio will be defined automatically to balanced
-            the dataset. Otherwise, the ratio will corresponds to the number
-            of samples in the minority class over the the number of samples
-            in the majority class.
-
-        random_state : int, RandomState or None, optional (default=None)
-
-            - If int, random_state is the seed used by the random number
-            generator;
-            - If RandomState instance, random_state is the random number
-            generator;
-            - If None, the random number generator is the RandomState instance
-            used by np.random.
-
-        Returns
-        -------
-        None
-
-        """
-        self.ratio = ratio
-        self.random_state = random_state
-        self.logger = logging.getLogger(__name__)
-
-    def fit(self, X, y):
-        """Find the classes statistics before to perform sampling.
-
-        Parameters
-        ----------
-        X : ndarray, shape (n_samples, n_features)
-            Matrix containing the data which have to be sampled.
-
-        y : ndarray, shape (n_samples, )
-            Corresponding label for each sample in X.
-
-        Returns
-        -------
-        self : object,
-            Return self.
-
-        """
-
-        super(BaseBinarySampler, self).fit(X, y)
-
-        # Check that the target type is binary
-        if not type_of_target(y) == 'binary':
-            warnings.simplefilter('always', UserWarning)
-            warnings.warn('The target type should be binary.')
+class BaseSampler(SamplerMixin):
```

Comment:
I am not in favor. I would like the sampler to have a `SamplerMixin` as the `ClassifierMixin` or `TransformerMixin`